components/feature/prompts/src/main/java/mozilla/components/feature/prompts/AlertDialogFragment.kt

```diff
@@ -0,0 +1,101 @@
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.support.v7.app.AlertDialog
+import android.view.LayoutInflater
+import android.widget.CheckBox
+
+private const val KEY_MANY_ALERTS = "KEY_MANY_ALERTS"
+private const val KEY_USER_CHECK_BOX = "KEY_USER_CHECK_BOX"
+
+/**
+ * [android.support.v4.app.DialogFragment] implementation to display web Alerts with native dialogs.
+ */
+internal class AlertDialogFragment : PromptDialogFragment() {
+
+    /**
+     * Tells if a checkbox should be shown for preventing this [sessionId] for showing more dialogs.
+     */
+    private val manyDialogs: Boolean by lazy { safeArguments.getBoolean(KEY_MANY_ALERTS) }
+
+    /**
+     * Stores the user's decision from the checkbox
+     * for preventing this [sessionId] for showing more dialogs.
+     */
+    private var userSelectionNoMoreDialogs: Boolean
+        get() = safeArguments.getBoolean(KEY_USER_CHECK_BOX)
+        set(value) {
+            safeArguments.putBoolean(KEY_USER_CHECK_BOX, value)
+        }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val builder = AlertDialog.Builder(requireContext())
+            .setTitle(title)
+            .setCancelable(true)
+            .setMessage(message)
+            .setPositiveButton(android.R.string.ok) { _, _ ->
+                onPositiveClickAction()
+            }
+        return (if (manyDialogs) addCheckbox(builder) else builder)
+            .create()
+    }
+
+    override fun onCancel(dialog: DialogInterface?) {
+        super.onCancel(dialog)
+        feature?.onCancel(sessionId)
+    }
+
+    private fun onPositiveClickAction() {
+        if (!userSelectionNoMoreDialogs) {
+            feature?.onCancel(sessionId)
+        } else {
+            feature?.onShouldMoreDialogsChecked(sessionId, userSelectionNoMoreDialogs)
+        }
+    }
+
+    @SuppressLint("InflateParams")
+    private fun addCheckbox(builder: AlertDialog.Builder): AlertDialog.Builder {
+        val inflater = LayoutInflater.from(requireContext())
+        val view = inflater.inflate(R.layout.mozac_feature_many_dialogs_checkbox_dialogs, null)
+        val checkBox = view.findViewById<CheckBox>(R.id.no_more_dialogs_check_box)
+        checkBox.setOnCheckedChangeListener { _, isChecked ->
+            userSelectionNoMoreDialogs = isChecked
+        }
+        builder.setView(view)
+
+        return builder
+    }
+
+    companion object {
+        /**
+         * A builder method for creating a [AlertDialogFragment]
+         * @property sessionId to create the dialog.
+         * @property title the title of the dialog.
+         * @property message the message of the dialog.
+         * @property manyDialogs tells if this [sessionId] has shown many dialogs is short period of time,
+         * if is true a checkbox will be part of the dialog, for the user to choose
+         * if wants to prevent this [sessionId] continuing showing dialogs.
```

Comment:
Hmm this seems like an unneeded indirection. GeckoView tells us whether or not to display a Checkbox, which we turn into `manyDialogs`, just to document that it's used to decide whether or not to display a checkbox.

Maybe just use rename to `showMoreDialogsCheckbox`?

Reply:
What about hasShownManyDialogs or ShouldShowCheckBox?